autocertkit/utils.py

```diff
@@ -1914,54 +1908,65 @@ def __call__(self, *args):
             signal.signal(signal.SIGALRM, old)
         return result
 
-def xml_to_dicts(xml, tag):
-    doc = minidom.parseString(xml.strip())
-    try:
-        result = []
-        elems = doc.getElementsByTagName(tag)
-        for el in elems:
-            # Note that we have to convert this dictionary to non-unicode
-            # strings, because we're being casual elsewhere.  That's why we're
-            # not just returning dict(el.attributes.items()).
-            item = {}
-            for k, v in el.attributes.items():
-                item[str(k)] = str(v)
-            result.append(item)
-    finally:
-        doc.unlink()
-    log.debug(result)
-    return result
+def json_loads(json_data):
```

Comment:
1. Why space in the key problem?

> This change is to prevent regression. 
> list_to_xml() method in ack plugin used to convert all whitespaces to _ so that xml tag will not have whitespace in name. therefore ACK kit is written accordingly to refer each key with _ and not whitespace.
> 
> Json has no such limitation though using whitespace in key means changing ACK at many places where it is referring to key .
> For example "Kernel name" became "Kernel_name" and now ACK uses it at many places.
> 

> ``` bash
> auto-cert-kit/autocertkit# grep -r "Kernel_name"
> utils.py:    ifaces = [dev['Kernel_name'] for dev in devices if device['PCI_id'] == dev['PCI_id']]
> utils.py:        ifaces.append(device['Kernel_name'])
> test_generators.py:            if device_rec['Kernel_name'] == self.interface:
> ack_cli.py:                      if dev['Kernel_name'] in ifaces_to_test]
> ack_cli.py:        ifaces.append(dev['Kernel_name'])
> testbase.py:        log.debug("Equivalent devices for %s: %s" % (self.config['device_config']['Kernel_name'],
> ```

2. Why do we need to parse when value is list or dict?

> we want to convert each unicode string to utf-8 encoded str type.
> For dict, we also need to ensure keys are following same naming as earlier(when using xml), to prevent regression.

3. What is source of unicode? I thought we all use utf-8 everywhere in XS.

> Json introduces unicode, while converting json string to python object. hence need to explicitly convert all unicode to utf-8 encoding by putting a str type conversion.